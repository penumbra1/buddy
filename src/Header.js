import React from "react";
import { Link } from "@reach/router";
import styled from "@emotion/styled";
import Logo from "./logo.svg";
import Search from "./icon-search.svg";
import Heart from "./icon-heart.svg";
import colors from "./colors";

const Container = styled.header`
  background-color: ${colors.dark};
  position: sticky;
  top: 0;
  z-index: 1;
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-bottom: 20px;
  padding: 1rem 2rem;
`;

const Nav = styled.nav`
  display: flex;
  flex-wrap: wrap;
  justify-content: flex-end;
  max-width: 30vw;
`;

const IconContainer = styled.div`
  width: ${props => props.width || "36px"};
  fill: ${props => props.fill || colors.greyLight};
  margin: 0.8rem;
  transition: fill 0.2s ease-in-out;

  &:hover {
    fill: ${colors.primary};
  }
`;

const IconLink = ({ image, description, to, ...props }) => (
  <Link to={to}>
    <span className="visuallyhidden">{description}</span>
    <IconContainer {...props} aria-hidden={true}>
      {image}
    </IconContainer>
  </Link>
);

const Header = () => {
  return (
    <Container>
      <IconLink
        to="/"
        image={<Logo />}
        width="120px"
        fill={colors.primary}
        description="Buddy"
      />
      <Nav>
        <IconLink to="/search-params" image={<Search />} description="Search" />
        <IconLink to="/favorites" image={<Heart />} description="Favorites" />
      </Nav>
    </Container>
  );
};

export default Header;
