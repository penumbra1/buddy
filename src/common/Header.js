import React from "react";
import { Link } from "@reach/router";
import styled from "@emotion/styled";
import Logo from "../assets/logo.svg";
import Search from "../assets/search.svg";
import Heart from "../assets/heart-line.svg";
import colors from "../colors";

const Wrapper = styled.header`
  background-color: ${colors.dark};
  position: sticky;
  top: 0;
  z-index: 1;
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-bottom: 2rem;
  padding: 1rem 5vw;
  border-radius: 0% 0% 15px 15px;
`;

const Nav = styled.nav`
  display: flex;
  flex-wrap: wrap;
  justify-content: flex-end;
  max-width: 30vw;
`;

const IconLink = ({ image, description, to, ...props }) => (
  <Link {...props} to={to}>
    <span className="visuallyhidden">{description}</span>
    <div aria-hidden={true}>{image}</div>
  </Link>
);

const StyledIconLink = styled(IconLink)`
  fill: ${props => props.fill || colors.greyLight};
  box-sizing: content-box;
  width: ${props => props.width || "3.6rem"};
  margin: 0 0.6rem;
  padding: 1.2rem;
  transition: fill 0.15s ease-in-out;

  div svg {
    display: block;
  }

  &:hover,
  &:focus {
    fill: ${colors.primaryLight};
  }
`;

const Header = () => {
  return (
    <Wrapper>
      <StyledIconLink
        to="/"
        image={<Logo />}
        width="13rem"
        fill={colors.primaryLight}
        description="Buddy"
      />
      <Nav>
        <StyledIconLink
          to="/search-params"
          image={<Search />}
          description="Search"
        />
        <StyledIconLink
          to="/favorites"
          image={<Heart />}
          description="Favorites"
          width="3.4rem"
        />
      </Nav>
    </Wrapper>
  );
};

export default Header;
