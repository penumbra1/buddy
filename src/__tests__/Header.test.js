import React from "react";
import { create } from "react-test-renderer";
import Header from "../Header";

test("should render correctly", () => {
  const c = create(<Header />).toJSON();
  expect(c).toMatchSnapshot();
});
