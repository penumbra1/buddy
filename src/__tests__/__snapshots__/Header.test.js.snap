// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
.emotion-4 {
  background-color: #393a3f;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 20px;
  padding: 1rem 2rem;
}

.emotion-0 {
  width: 120px;
  fill: #e8aa8c;
  margin: 0.8rem;
  -webkit-transition: fill 0.2s ease-in-out;
  transition: fill 0.2s ease-in-out;
}

.emotion-0:hover {
  fill: #e8aa8c;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  max-width: 30vw;
}

.emotion-1 {
  width: 36px;
  fill: #e2dcd5;
  margin: 0.8rem;
  -webkit-transition: fill 0.2s ease-in-out;
  transition: fill 0.2s ease-in-out;
}

.emotion-1:hover {
  fill: #e8aa8c;
}

<header
  className="emotion-4"
>
  <a
    aria-current="page"
    href="/"
    onClick={[Function]}
  >
    <span
      className="visuallyhidden"
    >
      Buddy
    </span>
    <div
      aria-hidden={true}
      className="emotion-0"
      fill="#e8aa8c"
      width="120px"
    >
      <SvgMock />
    </div>
  </a>
  <nav
    className="emotion-3"
  >
    <a
      href="/search-params"
      onClick={[Function]}
    >
      <span
        className="visuallyhidden"
      >
        Search
      </span>
      <div
        aria-hidden={true}
        className="emotion-1"
        width="36px"
      >
        <SvgMock />
      </div>
    </a>
    <a
      href="/favorites"
      onClick={[Function]}
    >
      <span
        className="visuallyhidden"
      >
        Favorites
      </span>
      <div
        aria-hidden={true}
        className="emotion-1"
        width="36px"
      >
        <SvgMock />
      </div>
    </a>
  </nav>
</header>
`;
